src/libraries/System.Security.Cryptography.Encoding/tests/Cbor.Tests/CborReaderTests.cs

```diff
@@ -12,51 +12,36 @@ namespace System.Security.Cryptography.Encoding.Tests.Cbor
-            {
-                var reader = new CborValueReader(ReadOnlySpan<byte>.Empty);
-                reader.Peek();
-            });
-        }
-
-        [Fact]
-        public static void TryPeek_EmptyBuffer_ShouldReturnFalse()
-        {
-            var reader = new CborValueReader(ReadOnlySpan<byte>.Empty);
-            bool result = reader.TryPeek(out CborInitialByte _);
-            Assert.False(result);
+            var reader = new CborReader(ReadOnlyMemory<byte>.Empty);
+            Assert.Equal(CborReaderState.EOF, reader.Peek());
         }
 
         [Theory]
-        [InlineData(0)]
-        [InlineData(1)]
-        [InlineData(10)]
-        [InlineData(100)]
-        [InlineData(255)]
-        public static void Peek_SingleByteBuffer_ShouldReturnSameByte(byte initialByte)
+        [InlineData(CborMajorType.UnsignedInteger, CborReaderState.UnsignedInteger)]
+        [InlineData(CborMajorType.NegativeInteger, CborReaderState.NegativeInteger)]
+        [InlineData(CborMajorType.ByteString, CborReaderState.ByteString)]
+        [InlineData(CborMajorType.TextString, CborReaderState.TextString)]
+        [InlineData(CborMajorType.Array, CborReaderState.StartArray)]
+        [InlineData(CborMajorType.Map, CborReaderState.StartMap)]
+        [InlineData(CborMajorType.Tag, CborReaderState.Tag)]
+        [InlineData(CborMajorType.Special, CborReaderState.Special)]
+        internal static void Peek_SingleByteBuffer_ShouldReturnExpectedState(CborMajorType majorType, CborReaderState expectedResult)
         {
-            ReadOnlySpan<byte> buffer = stackalloc byte[] { initialByte };
-            var reader = new CborValueReader(buffer);
-            CborInitialByte header = reader.Peek();
-            Assert.Equal(initialByte, header.InitialByte);
+            ReadOnlyMemory<byte> buffer = new byte[] { (byte)((byte)majorType << 5) };
+            var reader = new CborReader(buffer);
+            Assert.Equal(expectedResult, reader.Peek());
         }
 
-        [Theory]
-        [InlineData(0)]
-        [InlineData(1)]
-        [InlineData(10)]
-        [InlineData(100)]
-        [InlineData(255)]
-        public static void TryPeek_SingleByteBuffer_ShouldReturnSameByte(byte initialByte)
+        [Fact]
+        public static void CborReader_ReadingTwoPrimitiveValues_ShouldThrowInvalidOperationException()
         {
-            ReadOnlySpan<byte> buffer = stackalloc byte[] { initialByte };
-            var reader = new CborValueReader(buffer);
-            bool result = reader.TryPeek(out CborInitialByte header);
-
-            Assert.True(result);
-            Assert.Equal(initialByte, header.InitialByte);
+            ReadOnlyMemory<byte> buffer = new byte[] { 0 };
+            var reader = new CborReader(buffer);
+            reader.ReadInt64();
+            Assert.Equal(CborReaderState.Finished, reader.Peek());
+            Assert.Throws<InvalidOperationException>(() => reader.ReadInt64());
```

Comment:
This test is ambiguous, because the buffer is also out of data.  Perhaps make the input be `{ 0, 1 }`.

If someone has the middle of a payload and wants to read everything, can they do it? (Can they determine the amount of input buffer length that gets consumed per operation?)